### DISCUSSION TITLE
High level system overview?

### DISCUSSION THREAD

**Discussion by theoephraim** [vuejs/language-tools#945] [General]
Would love to help contribute to volar and get more involved, but a big blocker is that I feel a bit lost as to how the different components all work together. Would be super handy to have some kind of primer on where responsibilities start and stop. For example, running into some obscure type errors, it's unclear if these are issues with vue/core, vue-tsc, typescript-vue-plugin, etc...

A sort of high level overview / diagram of how all these pieces work together would be very helpful, and I think could hopefully allow others to help out more easily.

Thanks so much for all your work!

**Comment by johnsoncodehk** [comment]
Hi @theoephraim, I planned to this after refactored, because the project architecture will be modified a lot. Thank you. :)

**Comment by johnsoncodehk** [comment]
See #1044.

**Comment by sevillaarvin** [comment]
Hi @theoephraim, did you ever figure it out? I can connect to the server from the git repo, but I don't know how to debug it. I'm connecting my emacs/eglot via `/path/to/volar/packages/vue-language-server/bin/vue-language-server.js --stdio`, but I would like to print `console.log`s to debug inside volar.
